smartdashboard/views.py

```diff
@@ -341,3 +345,467 @@ def __init__(
+        return pd.DataFrame(columns=self.columns)
+
+    def _get_data_file(self) -> str:
+        """On click event to return csv data for the export button
+
+        :return: CSV data
+        :rtype: str
+        """
+        if self.telemetry:
+            try:
+                return pd.read_csv(self.files.graph_file).to_csv()
+            except FileNotFoundError:
+                self.table_element.info(self.message)
+        return ""
+
+
+class MemoryView(DualView):
+    """View class for memory section of the Database Telemetry page"""
+
+    @property
+    def files(self) -> Files:
+        """Returns a tuple of the files used for telemetry"""
+        if self.shard is not None:
+            return Files(self.shard.memory_file, self.shard.memory_file)
+
+        return Files("", "")
+
+    @property
+    def message(self) -> str:
+        """Returns the error message if files cannot be found"""
+        if self.shard is not None:
+            return f"Memory information could not be found for {self.shard.name}"
+
+        return ""
+
+    @property
+    def columns(self) -> t.List[str]:
+        """Returns columns for the graph dataframe"""
+        return ["timestamp", "used_memory", "used_memory_peak", "total_system_memory"]
+
+    def enable_export_button(self) -> None:
+        """Create an export data button"""
+        if self.shard is not None and self.telemetry:
+            try:
+                self.export_button.download_button(
+                    label="Export Data",
+                    data=self._get_data_file(),
+                    file_name=f"{self.shard.name}-memory.csv",
+                    mime="text/csv",
+                    key="memory",
+                )
+            except FileNotFoundError:
+                self.export_button.empty()
+        else:
+            self.export_button.empty()
+
+    def initial_load(self) -> None:
+        """Updates the table and graph with initial dataframes"""
+        table_df = self.telemetry_df.copy(deep=True)
+        table_df = self.process_dataframe(table_df)
+        graph_df: pd.DataFrame = table_df.copy(deep=True)
+        if self.telemetry_df.shape[0] >= self.window_size:
+            graph_df = graph_df.sample(self.window_size)
+            self.sampling = True
+        self._update_graph(graph_df)
+        self._update_table(table_df)
+
+    def _handle_data(self, graph_delta_df: pd.DataFrame) -> None:
+        """Updates the table and graph with appropriate dataframes"""
+        table_df = self.telemetry_df.copy(deep=True)
+        graph_df: pd.DataFrame = self.telemetry_df.copy(deep=True)
+        if graph_df.shape[0] >= self.window_size:
+            graph_df = graph_df.sample(self.window_size)
+            self.sampling = True
+            self._update_graph(self.process_dataframe(graph_df))
+        else:
+            self._update_graph(self.process_dataframe(graph_delta_df))
+        self._update_table(self.process_dataframe(table_df))
```

Comment:
I've figured out what I can do! I can get rid of `initial_load` by sending in `self.telemetry_df` into `_handle_data`. It will still sample if it needs to, or graph the delta, which just so happens to be the entire dataframe initially.